### ISSUE TITLE
TextField contentPadding + Expanded error

### ISSUE THREAD

**Issue by v7lin** [flutter/flutter#46964]
## Steps to Reproduce

If i don't set contentPadding，it's work right.

[preview image](https://github.com/v7lin/bug-report/blob/master/Screenshot_20191213_174521_com.likereading.xiyue.jpg)

```dart
Widget buildSearchBar() {
    return Container(
      decoration: BoxDecoration(
          color: ThemeColor.paper, borderRadius: BorderRadius.circular(15)),
      height: 30,
      child: Row(
        children: <Widget>[
          SizedBox(width: 10),
          Image.asset(
            'img/common/search_icon.png',
            color: Color(0xFFC8C8C8),
            width: 13.0,
            height: 13.0,
          ),
          SizedBox(width: 5),
          Expanded(
            child: TextField(
              autofocus: true,
              controller: _searchController,
              textInputAction: TextInputAction.search,
              inputFormatters: <TextInputFormatter>[
                LengthLimitingTextInputFormatter(50)
              ],
              style: TextStyle(fontSize: 14, color: ThemeColor.darkGray),
              decoration: InputDecoration(
                hintText:
                    localize(_hotKeys != null ? _hotKeys.first : '请输入关键字搜索'),
                hintStyle: TextStyle(color: ThemeColor.gray, fontSize: 14.0),
                border: InputBorder.none,
                contentPadding: EdgeInsets.only(left: 3.0, right: 8.0),
              ),
              onChanged: (String text) {
                if (text.isEmpty) {
                  _associationalWords = [];
                  _searchStatus = SearchStatus.blank;
                } else {
                  _searchStatus = SearchStatus.associational;
                  _subject.add(text);
                }
                setState(() {});
              },
              onSubmitted: (String text) {
                Analytics.event('dd47938eec46a6fd08f9a571d3e60f60');
                search();
              },
              cursorColor: ThemeColor.darkGray,
              cursorWidth: 1.0,
            ),
          ),
          _searchController.text.isEmpty
              ? SizedBox()
              : GestureDetector(
                  onTap: () {
                    _searchController.text = '';
                    _searchStatus = SearchStatus.blank;
                    setState(() {});
                  },
                  behavior: HitTestBehavior.opaque,
                  child: Image.asset('img/common/text_field_clean.png',
                      width: 30, height: 30),
                ),
        ],
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: createAppBar(
          context: context,
          titleWidget: buildSearchBar(),
          titleSpacing: 0,
          actions: <Widget>[
            Button(
              width: 60.0,
              height: kToolbarHeight,
              margin: EdgeInsets.only(right: 2.0),
              onPressed: search,
              child: createText('搜索',
                  style: TextStyle(color: ThemeColor.primary, fontSize: 16)),
            ),
          ]),
    );
  }
```

**Target Platform:** Android
**Target OS version/browser:** Android Q
**Devices:** HUAWEI VCE AL00

## Logs

```
```

```
No issues found! (ran in 13.2s)
```

```
[✓] Flutter (Channel stable, v1.12.13+hotfix.5, on Mac OS X 10.14.6 18G87,
    locale zh-Hans-CN)
    • Flutter version 1.12.13+hotfix.5 at /Users/v7lin/flutter
    • Framework revision 27321ebbad (2 days ago), 2019-12-10 18:15:01 -0800
    • Engine revision 2994f7e1e6
    • Dart version 2.7.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/v7lin/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling
      support)
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = /Users/v7lin/Library/Android/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.2, Build version 11B52
    • CocoaPods version 1.8.4

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 41.1.2
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)

[!] Android Studio
    • Android Studio at /Applications/Android Studio 4.0 Preview.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)

[✓] IntelliJ IDEA Community Edition (version 2019.1.4)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 36.0.4
    • Dart plugin version 191.8423

[!] VS Code (version 1.40.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    ✗ Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (1 available)
    • VCE AL00 • JPFDU19523006521 • android-arm64 • Android 9 (API 28)

! Doctor found issues in 2 categories.
```

**Comment by iapicca** [comment]
Hi @v7lin 
the issue seems to occur 
when `expanded` and `contentPadding`
are used at the same time

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) => MaterialApp(
    home: MyHomePage(),
  );
}

class MyHomePage extends StatelessWidget {
  @override
  Widget build(BuildContext context) => Scaffold(
    body: Center(
      //child: Expanded(
        child: TextField(
          textInputAction: TextInputAction.search,
          decoration: InputDecoration(
            contentPadding: EdgeInsets.only(left: 3.0, right: 8.0),
          ),
        //),
  ),),);
}
```

**Comment by v7lin** [comment]
it has same error with

```dart
  Widget buildSearchBar() {
    return Container(
      decoration: BoxDecoration(
          color: ThemeColor.paper, borderRadius: BorderRadius.circular(15)),
      height: 30,
      padding: EdgeInsets.only(left: 10),
      child: Stack(
        children: <Widget>[
          Align(
            alignment: Alignment.centerLeft,
            child: Image.asset(
              'img/common/search_icon.png',
              color: Color(0xFFC8C8C8),
              width: 13.0,
              height: 13.0,
            ),
          ),
          Container(
            color: Colors.red,
            margin: EdgeInsets.only(left: 18),
            child: TextField(
              autofocus: true,
              controller: _searchController,
              textInputAction: TextInputAction.search,
              inputFormatters: <TextInputFormatter>[
                LengthLimitingTextInputFormatter(50)
              ],
              style: TextStyle(fontSize: 14, color: ThemeColor.darkGray),
              decoration: InputDecoration(
                hintText:
                    localize(_hotKeys != null ? _hotKeys.first : '请输入关键字搜索'),
                hintStyle: TextStyle(color: ThemeColor.gray, fontSize: 14.0),
                border: InputBorder.none,
                contentPadding: EdgeInsets.only(left: 3.0, right: 8.0),
              ),
              onChanged: (String text) {
                if (text.isEmpty) {
                  _associationalWords = [];
                  _searchStatus = SearchStatus.blank;
                } else {
                  _searchStatus = SearchStatus.associational;
                  _subject.add(text);
                }
                setState(() {});
              },
              onSubmitted: (String text) {
                Analytics.event('dd47938eec46a6fd08f9a571d3e60f60');
                search();
              },
              cursorColor: ThemeColor.darkGray,
              cursorWidth: 1.0,
            ),
          ),
          Align(
            alignment: Alignment.centerRight,
            child: _searchController.text.isEmpty
                ? SizedBox()
                : GestureDetector(
              onTap: () {
                _searchController.text = '';
                _searchStatus = SearchStatus.blank;
                setState(() {});
              },
              behavior: HitTestBehavior.opaque,
              child: Image.asset('img/common/text_field_clean.png',
                  width: 30, height: 30),
            ),
          ),
        ],
      ),
    );
  }
```

**Comment by v7lin** [comment]
text will be cut

[preview error image](https://github.com/v7lin/bug-report/blob/master/Screenshot_20191216_093632_com.likereading.xiyue.jpg)

```
  Widget buildSearchBar() {
    return Container(
      decoration: BoxDecoration(
          color: ThemeColor.paper, borderRadius: BorderRadius.circular(15)),
      height: 30,
      child: Row(
        children: <Widget>[
          SizedBox(width: 10),
          Image.asset(
            'img/common/search_icon.png',
            color: Color(0xFFC8C8C8),
            width: 13.0,
            height: 13.0,
          ),
          SizedBox(width: 5),
          Expanded(
            child: Container(
              padding: EdgeInsets.only(left: 3.0, right: 8.0),
              child: TextField(
                autofocus: true,
                controller: _searchController,
                textInputAction: TextInputAction.search,
                inputFormatters: <TextInputFormatter>[
                  LengthLimitingTextInputFormatter(50)
                ],
                style: TextStyle(fontSize: 14, height: 1.0, color: ThemeColor.darkGray),
                maxLines: 1,
                minLines: 1,
                decoration: InputDecoration(
                  hintText:
                  localize(_hotKeys != null ? _hotKeys.first : '请输入关键字搜索'),
                  hintStyle: TextStyle(color: ThemeColor.gray, fontSize: 14.0, height: 1.0),
                  border: InputBorder.none,
//                  contentPadding: EdgeInsets.only(left: 3.0, right: 8.0),
                ),
                onChanged: (String text) {
                  if (text.isEmpty) {
                    _associationalWords = [];
                    _searchStatus = SearchStatus.blank;
                  } else {
                    _searchStatus = SearchStatus.associational;
                    _subject.add(text);
                  }
                  setState(() {});
                },
                onSubmitted: (String text) {
                  Analytics.event('dd47938eec46a6fd08f9a571d3e60f60');
                  search();
                },
                cursorColor: ThemeColor.darkGray,
                cursorWidth: 1.0,
              ),
            ),
          ),
          _searchController.text.isEmpty
              ? SizedBox()
              : GestureDetector(
                  onTap: () {
                    _searchController.text = '';
                    _searchStatus = SearchStatus.blank;
                    setState(() {});
                  },
                  behavior: HitTestBehavior.opaque,
                  child: Image.asset('img/common/text_field_clean.png',
                      width: 30, height: 30),
                ),
        ],
      ),
    );
  }
```

**Comment by justinmc** [comment]
Have you tried using `isDense`?

Just mentioning it because this issue was linked from https://github.com/flutter/flutter/issues/44726 which had that solution.  TextField height was increased recently (in https://github.com/flutter/flutter/pull/42449).

**Comment by v7lin** [comment]
yes

**Comment by justinmc** [comment]
Alright then I'll try to reproduce this.  Would you be able to paste a minimal app that reproduces the problem so I can try it out?

**Comment by caseyryan** [comment]
The same here. I used 1.10 before bu today I've updated it to v1.12.13+hotfix.5 and got exactly the same problem with all my TextFields. 
I'm using InputDecoration like this:
```dart
decoration: InputDecoration(
  alignLabelWithHint: true,
//  contentPadding: EdgeInsets.only(top: 1),
  border: InputBorder.none,
  hintText: widget.searchTitle,
),
```

So if I comment this line out
```dart
//  contentPadding: EdgeInsets.only(top: 1),
```
everything is fine. With this line (regardless of padding value), the text is aligned by bottom, just like a topic starter showed in his screenshot
